CUJExecutor/buildProfile

```diff
@@ -0,0 +1,117 @@
+STASHING_CREATED_PROFILE="/tmp/build_created_profile.txt"
+
+echo "" > $COMMAND_SUMMARY
+
+
+
+
+
+
+echo "Resetting the profile...."
+adb shell cmd package compile --reset com.google.android.apps.maps
+
+echo "Setting the base profile..."
+#If we're given a file name, set that to be the profile we use and exit
+if [ -f "$COMPILATION_FLAG_SEQ" ] 
+then 
+	
+	FILE_EXTENSION="${COMPILATION_FLAG_SEQ##*.}"
+	echo $FILE_EXTENSION
+	if [ "$FILE_EXTENSION" = "prof" ]
+	then
+		cp $COMPILATION_FLAG_SEQ $STASHING_CREATED_PROFILE
+		echo -n "Base: $COMPILATION_FLAG_SEQ, Actions:" > $STASHING_COMPILED_ACTIONS
+		exit 0
+	fi
+	
+fi
+
+#Otherwise, depending on the flag specified by the first character, set the base profile and build the profile based on each subsequent profile
+if [ "$BASE_PROFILE" = "e" ]
+then
+	cp $EMPTY_PROFILE $STASHING_CREATED_PROFILE
+	echo -n "Base: Empty, Actions:" > $STASHING_COMPILED_ACTIONS
+elif [ "$BASE_PROFILE" = "u" ]
+then
+	cp $CLOUD_PROFILE $STASHING_CREATED_PROFILE	
+	echo -n "Base: Cloud, Actions:" > $STASHING_COMPILED_ACTIONS
+elif [ "$BASE_PROFILE" = "s" ]
+then
+	cp $STARTUP_PROFILE $STASHING_CREATED_PROFILE	
+	echo -n "Base: Startup, Actions:" > $STASHING_COMPILED_ACTIONS
+else
+	echo "ERROR: invalid base profile"
+	exit 1
+fi
+
+
+for ((i=1; i<${#COMPILATION_FLAG_SEQ}; i++))
+do
+	COMPILATION_FLAG="${COMPILATION_FLAG_SEQ:$i:1}"
+
+	if [ "$COMPILATION_FLAG" != "c" ] && [ "$COMPILATION_FLAG" != "p" ] && [ "$COMPILATION_FLAG" != "f" ] && [ "$COMPILATION_FLAG" != "m" ] 
+	then	
+		echo "Builder actions flag: $COMPILATION_FLAG is invalid"
+		exit 1
+	fi
+
+	if [ "$COMPILATION_FLAG" = "m" ] 
+	then 
+		
+		echo "Executing preparatory actions.."
+		./runCujNTimes $INPUT_FILE p 1
+
+		echo "Waiting for prepatory actions to be added to the profile..."
+		sleep 60
+	fi
+	
+
+	echo "Pushing our working profile"
+	adb push $STASHING_CREATED_PROFILE $CUR_PROFILE
+
+	echo "Restoring profile permissions"
+	adb shell chmod a+rw $CUR_PROFILE
+
+	adb logcat -c
+
+	echo "Executing the actions we'd like in the profile..."
+	./runCujNTimes $INPUT_FILE $COMPILATION_FLAG 1
+	
+	echo "Waiting for the desired actions to be added to the profile..."
+	sleep 60
+
+	if [ $i != "0" ] #For some reason, if we do this every time, cleanup actions will be included, even if a clear comes immediately after. An alternative would be to sleep before clearing.
+	then
+		echo "Verifying and logging which actions were executed"
+		COMPILED_ARR=$(adb logcat -s "actions-run" -v raw -d | sed -n -e '/^\[/p')
+		echo -n " ${COMPILED_ARR}" >> $STASHING_COMPILED_ACTIONS
+	fi
+
+	echo "Pulling the current profile"
+	adb pull $CUR_PROFILE $STASHING_CREATED_PROFILE
+
+	echo "Getting profile size..."
+	adb shell stat -c%s $CUR_PROFILE
+
+	echo "Finishing the CUJ..."
+	./runCujNTimes $INPUT_FILE ${COMPILATION_FLAG}r 1
+
+
+	echo "Force stopping Maps..." #Want to make sure that app is stopped when we mess with the profiles at start of the loop in callee
```

Comment:
Generally easier to read if comment is right above the code, as opposed to the right.